addons/binding/org.openhab.binding.rme/src/main/java/org/openhab/binding/rme/handler/RMEThingHandler.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) 2014 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.rme.handler;
+
+import static org.openhab.binding.rme.RMEBindingConstants.*;
+
+import java.math.BigDecimal;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RMEThingHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ * 
+ * @author Karel Goderis - Initial contribution
+ */
+public class RMEThingHandler extends SerialThingHandler {
+
+	private Logger logger = LoggerFactory.getLogger(RMEThingHandler.class);
+
+	public RMEThingHandler(Thing thing) {
+		super(thing);
+	}
+	
+	@Override
+	public void initialize() {
+		logger.debug("Initializing RME handler.");
+		
+		Configuration configuration = editConfiguration();
+		configuration.put(BAUD_RATE, (int) 2400);
+		configuration.put(BUFFER_SIZE, (int) 1024);
+		updateConfiguration(configuration);
+		
+		super.initialize();
+	}
+
+	@Override
+	public void handleCommand(ChannelUID channelUID, Command command) {
+		logger.warn("The RME Rain Manager is a read-only device and can not handle commands");
+	}
+
+	@Override
+	public void onDataReceived(String line) {
+
+		line = StringUtils.chomp(line);
+
+		// little hack to overcome Locale limits of the RME Rain Manager
+		// note to the attentive reader : should we add support for system
+		// locale's
+		// in the Type classes? ;-)
+		line = line.replace(",", ".");
+		line = line.trim();
+
+		Pattern RESPONSE_PATTERN = Pattern
+				.compile("(.*);(0|1);(0|1);(0|1);(0|1);(0|1);(0|1);(0|1);(0|1);(0|1)");
+
+		Matcher matcher = RESPONSE_PATTERN.matcher(line);
+		if (matcher.matches()) {
+			for (int i = 1; i <= matcher.groupCount(); i++) {
+				switch (DataField.get(i)) {
+				case LEVEL: {
+					DecimalType decimalType = new DecimalType(matcher.group(i));
+					updateState(new ChannelUID(getThing().getUID(), DataField
+							.get(i).channelID()), decimalType);
+					break;
+				}
+				case MODE: {
+					StringType stringType = null;
+					if (matcher.group(i).equals("0")) {
+						stringType = new StringType("Manual");
```

Comment:
It is probably more efficient to declare the StringTypes as a constant instead of instantiating new objects every time.